### PR TITLE
Fix debugfs audit capability service limit

### PR DESCRIPTION
**Pull request by ktsaou**
## Summary
- Allow `CAP_AUDIT_CONTROL` in the systemd `CapabilityBoundingSet` so `debugfs.plugin` can execute with its packaged audit capability.
- Document the `debugfs.plugin` audit capability requirement.
- Add runtime/package validation for installed plugin file capabilities versus the installed systemd unit.
- Add static archive validation for static installer-declared plugin capabilities versus the packaged systemd unit.

## Root Cause
`debugfs.plugin` is granted `cap_dac_read_search,cap_audit_control` by package/install paths, but the modern systemd unit constrained `CapabilityBoundingSet` without `CAP_AUDIT_CONTROL`. Linux capability semantics mask file capabilities outside the process bounding set during `execve()`, so the plugin can fail before its own code runs.

## Container And Static Install Check
- Static installs use the generated modern systemd unit, so they inherit the runtime fix; the new static archive guard catches future drift before publishing an archive.
- Docker packaging installs privileged plugins as setuid and launches Netdata directly, so there is no container-local systemd `CapabilityBoundingSet` to update.
- Docker/Kubernetes examples are left unchanged because adding `AUDIT_CONTROL` changes the documented container privilege profile and should be a separate product/security decision.

## Validation
- `sh -n packaging/runtime-check.sh`
- `shellcheck packaging/runtime-check.sh`
- `bash -n packaging/makeself/jobs/81-netdata-runtime-check.sh`
- `shellcheck -x packaging/makeself/jobs/81-netdata-runtime-check.sh`
- `systemd-analyze verify` on the generated service unit after substituting `@sbindir_POST@`
- fake installed-tree runtime positive/negative test for `CAP_AUDIT_CONTROL`
- fake static archive positive/negative test for `CAP_AUDIT_CONTROL`
- `git diff --check`

---
## Summary by cubic
Allow `CAP_AUDIT_CONTROL` in the Netdata systemd service so `debugfs.plugin` can run with its audit capability. Validation now detects capability mismatches at install time and in static archives, and checks installer/unit readability.

- **Bug Fixes**
  - Added `CAP_AUDIT_CONTROL` to `CapabilityBoundingSet` in `netdata.service` for `debugfs.plugin`.

- **New Features**
  - Validation: hardened `CapabilityBoundingSet` parsing (supports `~`, empty, multiline), `NETDATA_SYSTEMD_UNIT` override and common path search, skipping via `NETDATA_SKIP_LIBEXEC_PARTS`, safe no-op if unit or `getcap` is missing; static archive guard compares installer-declared caps to the packaged unit and verifies the installer/unit are readable; docs updated for `debugfs.plugin`.

<sup>Written for commit fa76e031352aeb57aa59095194e4b2dc32cf3943. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/netdata/netdata/pull/22831?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packaging/runtime-check.sh [hunk 1]
```diff
@@ -31,6 +31,94 @@ wait_for() {
...
+}
+
+find_netdata_systemd_unit() {
+    for unit in \
+        ${NETDATA_SYSTEMD_UNIT:-} \
+        /etc/systemd/system/netdata.service \
+        /usr/lib/systemd/system/netdata.service \
+        /lib/systemd/system/netdata.service \
+        /usr/local/lib/systemd/system/netdata.service; do
+        if [ -f "${unit}" ]; then
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply packaging/runtime-check.sh]: Valid. Fixed in b482e6f7c3: unit discovery now requires a readable regular file, and an explicit NETDATA_SYSTEMD_UNIT that is not readable fails validation instead of falling back to another unit. I also added a focused unreadable-unit smoke test during local validation.

File: packaging/runtime-check.sh [hunk 2]
```diff
@@ -31,6 +31,94 @@ wait_for() {
...
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+systemd_unit_capability_bounding_set() {
+    awk '
+        /^[[:space:]]*CapabilityBoundingSet[[:space:]]*=/ {
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply packaging/runtime-check.sh]: Valid. Fixed in b482e6f7c3: the check now evaluates each requested capability with systemd CapabilityBoundingSet semantics instead of flattening lines naively. It handles line continuations, repeated OR assignments, inverted '~' assignments, empty resets, and bare '~' full-set resets. I mirrored the same parser in the static archive guard and validated those cases locally.

**Comment by ktsaou** [comment]
@cubic-dev-ai please review again

File: packaging/makeself/jobs/81-netdata-runtime-check.sh [hunk 3]
```diff
@@ -11,6 +11,171 @@ dump_log() {
...
+}
+
+check_static_installer_systemd_capability_bounding_set() {
+  local installer="${NETDATA_MAKESELF_PATH}/install-or-update.sh"
+  local unit="${NETDATA_INSTALL_PATH}/usr/lib/netdata/system/systemd/netdata.service"
+  local cap=
+  local rc=
+  local success=0
+
+  [ -f "${installer}" ] || return 0
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply packaging/makeself/jobs/81-netdata-runtime-check.sh:146]: Valid. Fixed in fa76e03135: the static archive guard now requires install-or-update.sh to be readable when it exists, and fails validation if it is not. I also added a focused local smoke test that chmods the installer unreadable and verifies the guard fails instead of extracting an empty capability set.

**Comment by ktsaou** [comment]
@cubic-dev-ai please review again

File: system/systemd/netdata.service.in [hunk 4]
```diff
@@ -40,6 +40,8 @@ Nice=0
 CapabilityBoundingSet=CAP_DAC_OVERRIDE
 # is required for apps plugin
 CapabilityBoundingSet=CAP_DAC_READ_SEARCH
+# is required for debugfs plugin audit subsystem monitoring
+CapabilityBoundingSet=CAP_AUDIT_CONTROL
```
Comment (ralphm) [inline system/systemd/netdata.service.in:44]: Verified that adding this as an override to the service fixes the issue.

**Review by ralphm — APPROVED** [review]
